README.md

```diff
@@ -12,7 +12,8 @@ Variant's basic building blocks are:
 - `variant<Ts...>` - a type-safe representation for sum-types / discriminated unions
 - `recursive_wrapper<T>` - a helper type to represent recursive "tree-like" variants
 - `apply_visitor(visitor, myVariant)` - to invoke a custom visitor on the variant's underlying type
-- `get<T>()` - a function to directly unwrap a variant's underlying type
+- `bool is<T>()` - a function to test for a specific underlying type `T`
+- `T get<T>()` - a function to directly unwrap a variant's underlying type
```

Comment:
> Should we document all of variant's api here?

@daniel-j-h good questions!  I wasn't aware of the other parts of variant's API either.  Motivation for this PR was simply that I found myself wishing for an `is<T>()` method and assuming it didn't exist because it wasn't in the readme (and the header code itself isn't too easy to skim as a reference).